server/flyway/sql/V50__update_fam_user_constraint_use_guid.sql

```diff
@@ -0,0 +1,5 @@
+-- Update unique constraint in fam user table to be user type code and user guid
+
+DROP INDEX app_fam.fam_usr_uk;
+
+CREATE UNIQUE INDEX fam_usr_uk ON app_fam.fam_user(user_type_code, lower(user_guid));
```

Comment:
I feel it doesn't hurt to add lower here just to avoid human error, when someone try to add user using api call directly (unlikely happen though). I agree to remove the lower in lambda function

Reply:
Hi Catherine,
I am in doubt of IDIM GUID format, and feel uncertain what it could be.
If we are speaking from real GUID format, it is a hexdecimal format thus it is case-insensitive(if I understand correctly).
But, **IDIM GUID format** from doc says **32 alphanumeric characters**, not the real GUID format.
It says "**Case preserving**", so I might guess it does not just generate from `0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ` characters, but also `abcdefghijklmnopqrstuvwxyz` characters.
However, it then says "**By convention, this is upper case**", this is really **confusing statement for
developers**. Does IDIM just for now generate that GUID using `0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ` but reserve using `abcdefghijklmnopqrstuvwxyz` in future?

So, at the end, treating it as case-sensitive or case-insensitive (using lower/upper) **is the same (if their
"By convention statement" stays true**; and thus it might not hurt in our **ON CONFLICT** statement we use
**(user_type_code, lower(user_guid))**.

But still, I am really curious if user A has (GUID = '**123IAMUNIQUE456**') and suddenly in IDIM system it has
user B (GUID='**123IAMUNIQUe456**') or user C (GUID='**123iAMUNIQUE456**') or user xyz (GUID='**123iamunique456**').... and **is allowed** because it says it is"**Case preserving**", then we are going to treating them as the same user with big hard to spot problem.

Although I favor not to compare GUID with lower/upper as we don't know if it is still possible the GUID is case-sensitive, but I think I am ok for what you have here.

Wonder if @basilv  has more understanding of this IDIM GUID field and has other opinion.